misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkLeaseManager.kt

```diff
@@ -40,9 +43,7 @@ internal class ZkLeaseManager @Inject internal constructor(
 
   private val ownerName = cluster.snapshot.self.name
   private val actionQueue = LinkedBlockingQueue<() -> Unit>()
-  private val leaseCheckExecutor = Executors.newCachedThreadPool(ThreadFactoryBuilder()
-      .setNameFormat("lease-monitor-%d")
-      .build())
+  private val leasePollInterval: Duration = Duration.ofSeconds(1L)
```

Comment:
haven't been following too closely, is there any particular science around 1 second?

Reply:
`checkHeld()` will check the in-memory map of cluster members to figure out if it should own the lease before attempting to talk to ZK, so this is a pretty cheap check. It should happen as frequently as possible to ensure the lease is always held by some process. The map is updated by a cluster watch and should happen infrequently.

Reply:
thanks for the explanation! (was more wondering 1 second vs 100ms, etc)

Reply:
It could probably be 100ms, but I think we can tolerate ~1 second of having no one owning a lease? Service startup/shutdown is on order of seconds anyways, so an ephemeral ZK lease node could take a few seconds to get deleted after shutdown is triggered anyways.

Reply:
mm yeah, that's a good way to put it!